src/main/java/de/teamlapen/vampirism/player/vampire/VampirePlayer.java

```diff
@@ -1065,69 +1102,136 @@ private void biteBlock(@Nonnull BlockPos pos, @Nonnull IBlockState blockState, @
             blood = ExtendedCreature.get((EntityCreature) entity).onBite(this);
             saturationMod = ExtendedCreature.get((EntityCreature) entity).getBloodSaturation();
-        } else if (type == BITE_TYPE.SUCK_BLOOD_PLAYER || type == BITE_TYPE.SUCK_BLOOD_HUNTER_PLAYER) {
+        } else if (victim_bite_type == BITE_TYPE.SUCK_BLOOD_PLAYER || victim_bite_type == BITE_TYPE.SUCK_BLOOD_HUNTER_PLAYER) {
             blood = VampirePlayer.get((EntityPlayer) entity).onBite(this);
             saturationMod = VampirePlayer.get((EntityPlayer) entity).getBloodSaturation();
-            if (type == BITE_TYPE.SUCK_BLOOD_HUNTER_PLAYER) {
+            if (victim_bite_type == BITE_TYPE.SUCK_BLOOD_HUNTER_PLAYER) {
                 player.addPotionEffect(new PotionEffect(MobEffects.POISON, 15, 2));
             }
-        } else if (type == BITE_TYPE.SUCK_BLOOD) {
+        } else if (victim_bite_type == BITE_TYPE.SUCK_BLOOD) {
             blood = ((IBiteableEntity) entity).onBite(this);
             saturationMod = ((IBiteableEntity) entity).getBloodSaturation();
-        } else if (type == BITE_TYPE.ATTACK || type == BITE_TYPE.ATTACK_HUNTER) {
-            checkAttributes(VReference.biteDamage);
-            float damage = getSpecialAttributes().bat ? 0.1F : (float) player.getEntityAttribute(VReference.biteDamage).getAttributeValue();
-            entity.attackEntityFrom(DamageSource.causePlayerDamage(player), damage);
-            if (entity.isEntityUndead() && player.getRNG().nextInt(4) == 0) {
-                player.addPotionEffect(new PotionEffect(MobEffects.POISON, 60));
-            } else if (type == BITE_TYPE.ATTACK_HUNTER) {
-                if (entity instanceof EntityPlayer && ItemHunterCoat.isFullyEquipped((EntityPlayer) entity)) {
-                    player.attackEntityFrom(DamageSource.causeThornsDamage(entity), damage);
-                }
-            }
-            if (specialAttributes.poisonous_bite) {
-                entity.addPotionEffect(new PotionEffect(MobEffects.POISON, (int) (Balance.vps.POISONOUS_BITE_DURATION * 20 * (getSpecialAttributes().bat ? 0.2F : 1F)), 1));
-            }
-        } else if (type == BITE_TYPE.NONE) {
+        } else if (victim_bite_type == BITE_TYPE.NONE) {
             return;
         }
-        biteCooldown = Balance.vp.BITE_COOLDOWN;
         if (blood > 0) {
             drinkBlood(blood, saturationMod);
             //TODO player.addStat(Achievements.suckingBlood, 1);
             NBTTagCompound updatePacket = bloodStats.writeUpdate(new NBTTagCompound());
             updatePacket.setInteger(KEY_SPAWN_BITE_PARTICLE, entity.getEntityId());
             sync(updatePacket, true);
         }
-        if (type == BITE_TYPE.SUCK_BLOOD || type == BITE_TYPE.SUCK_BLOOD_CREATURE || type == BITE_TYPE.SUCK_BLOOD_PLAYER) {
+        if (victim_bite_type == BITE_TYPE.SUCK_BLOOD || victim_bite_type == BITE_TYPE.SUCK_BLOOD_CREATURE || victim_bite_type == BITE_TYPE.SUCK_BLOOD_PLAYER) {
```

Comment:
I don't know why I but the advancement trigger here before, I think it would be best to move it to the end of the `if(blood > 0)` condition, so we don't need to check the bite type again.